tutorials/02_add_transform_keys.ipynb

```diff
@@ -0,0 +1,347 @@
+    "                    interpolation='AUTO',\n",
+    "                )            \n",
+    "            )\n",
+    "        # The function `use_actor_with_keys` sets transform keys for the actor in the sequence.\n",
+    "        # The transform keys are only stored in the sequence.\n",
+    "        # When the sequence is closed, the actor's location, rotation and scale will be restored to its original state(without transform keys).\n",
+    "        seq.use_actor_with_keys(actor=actor, transform_keys=transform_keys[name])\n",
+    "        actor_idx += 1\n",
+    "    \n",
+    "    # Add transform keys to koupen_chan to make it always sit on the table\n",
+    "    koupen_chan_transform_keys = copy.deepcopy(transform_keys['table'])\n",
+    "    for keys in koupen_chan_transform_keys:\n",
+    "        keys.location = (keys.location[0], keys.location[1], keys.location[2] + furnitures['table'].dimensions[2] * 0.5)\n",
+    "        keys.scale = koupen_chan.scale\n",
+    "    # The function `use_actor_with_keys` sets transform keys for the actor in the sequence.\n",
+    "    # The transform keys are only stored in the sequence.\n",
+    "    # When the sequence is closed, the actor will be restored to its original state(without transform keys).\n",
+    "    seq.use_actor_with_keys(actor=koupen_chan, transform_keys=koupen_chan_transform_keys)\n",
+    "\n",
+    "    #####################\n",
+    "    ##### Rendering #####\n",
+    "    #####################\n",
+    "\n",
+    "    # Add a camera and make it look at the specified location\n",
+    "    camera_location = (1.5, -2.0, 3.0)\n",
+    "    camera_rotation = xf_runner.utils.get_rotation_to_look_at(location=camera_location, target=(1.5, 1.0, 1.0))\n",
+    "    seq.spawn_camera(location=camera_location, rotation=camera_rotation, fov=90)\n",
+    "\n",
+    "    # Add a render job to renderer\n",
+    "    # In render job, you can specify the output path, resolution, render passes, etc.\n",
+    "    # The ``output_path`` is the path to save the rendered data.\n",
+    "    # The ``resolution`` is the resolution of the rendered image.\n",
+    "    # The ``render_passes`` define what kind of data you want to render, such as img, depth, normal, etc.\n",
+    "    # and what kind of format you want to save, such as png, exr, etc.\n",
+    "    seq.add_to_renderer(\n",
+    "        output_path=f'./tutorial02/outputs/{engine}/',\n",
+    "        resolution=(1280, 720),\n",
+    "        render_passes=[RenderPass('img', 'png')],\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 5. Render\n",
+    "\n",
+    "The following code renders all the render jobs and save the images to the ``output_path`` set in ``seq.add_to_renderer`` above."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "xf_runner.render()"
```

Comment:
`## 5. Render` and `## 6. Close the engine` are duplicates in tutorial01. Please consider squashing them into one cell.

Reply:
Plus, consider renaming this tutorial to something like 02_randomization.

Reply:
1. fixed in ce589b6
2. fixed in 4f478f9